src/flb_log.c

```diff
@@ -62,10 +62,83 @@ static inline int consume_byte(flb_pipefd_t fd)
     return 0;
 }
 
+static inline int log_file_sz_max(off_t size, struct flb_log *log)
+{
+    double sz_mb;
+
+    sz_mb = size / (1024.00 * 1024.00);
+    if (sz_mb >= log->max_sz_mb) {
+        return FLB_TRUE;
+    }
+
+    return FLB_FALSE;
+}
+
+static inline void log_file_rotate(struct flb_log *log)
+{
+    char *backup_file_name = NULL;
+    char *src_file_name = NULL;
+    int ret = -1;
+    int i;
+
+    if (log->max_history == 0) {
+        /*
+         * no backups.
+         * remove the file since size has exceeded.
+         */
+        remove(log->out);
+        return;
+    }
+
+    /* max 2 digits for backup name suffix and 1 each for '.' and '\0' */
+    size_t filename_len = strlen(log->out) + 4;
```

Comment:
is this expected here?
```suggestion
    size_t filename_len = snprintf(NULL, 0, "%"PRIu64, log->out);
```